src/agent.rs

```diff
@@ -437,25 +440,35 @@ impl Discoverer for Public {
                     entry.get_mut().create_bridge(callback)
                 }
                 Entry::Vacant(entry) => {
-                    let slab_base: Shared<Slab<Callback<AGN::Output>>> =
+                    let slab: Shared<Slab<Callback<AGN::Output>>> =
                         Rc::new(RefCell::new(Slab::new()));
-                    let slab = slab_base.clone();
-                    let handler = move |data: Vec<u8>| {
-                        let msg = FromWorker::<AGN::Output>::unpack(&data);
-                        match msg {
-                            FromWorker::WorkerLoaded => {
-                                // TODO Use `AtomicBool` lock to check its loaded
-                                // TODO Send `Connected` message
-                            }
-                            FromWorker::ProcessOutput(id, output) => {
-                                let callback = slab.borrow().get(id.raw_id()).cloned();
-                                if let Some(callback) = callback {
-                                    callback.emit(output);
-                                } else {
-                                    warn!(
-                                        "Id of handler for remote worker not exists <slab>: {}",
-                                        id.raw_id()
-                                    );
+                    let handler = {
+                        let slab = slab.clone();
+                        move |data: Vec<u8>, worker: Value| {
+                            let msg = FromWorker::<AGN::Output>::unpack(&data);
+                            match msg {
+                                FromWorker::WorkerLoaded => {
```

Comment:
You should also make sure to cleanup the LOADED and MSGS_QUEUE maps when the worker is terminated. Check out `Drop for PublicBridge` 👍

Reply:
Ive checked this situation a bit earlier, when worker drops there is nothing in LOADED map.
Its because queue is drained right after worker is online and never used for this worker after.

Reply:
@serzhiio I don't understand, how about this:

1. Worker is spawned
1. Worker receives message and puts it into msg queue
1. Worker loaded
1. Worker drains msg queue
1. Worker is dropped

In this case, the msg queue will be empty but the LOADED map will still have an entry for this worker. We should remove that entry when the worker is dropped.

Reply:
For purity of dataflow we can clean LOADED map, but since Remote agents have non unique its not very important.

Reply:
I think it is important, because if worker is dropped, and then spawned later, it will seem to already be loaded